### PR TITLE
Add sieve generator

### PR DESCRIPTION
**Pull request by pvcarrera**
Sieve is the list of exercises that need a generator #396. 

When #451 gets merge, I'll rebase this PR and remove the `bin/generate-sieve` file.

### CODE REVIEW

File: exercises/sieve/sieve_test.rb [hunk 1]
```diff
@@ -3,47 +3,54 @@
...
-      601, 607, 613, 617, 619, 631, 641, 643, 647, 653, 659, 661, 673,
-      677, 683, 691, 701, 709, 719, 727, 733, 739, 743, 751, 757, 761,
-      769, 773, 787, 797, 809, 811, 821, 823, 827, 829, 839, 853, 857,
-      859, 863, 877, 881, 883, 887, 907, 911, 919, 929, 937, 941, 947,
-      953, 967, 971, 977, 983, 991, 997
-    ]
+  def test_find_primes_up_to_1000
+    expected = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199, 211, 223, 227, 229, 233, 239, 241, 251, 257, 263, 269, 271, 277, 281, 283, 293, 307, 311, 313, 317, 331, 337, 347, 349, 353, 359, 367, 373, 379, 383, 389, 397, 401, 409, 419, 421, 431, 433, 439, 443, 449, 457, 461, 463, 467, 479, 487, 491, 499, 503, 509, 521, 523, 541, 547, 557, 563, 569, 571, 577, 587, 593, 599, 601, 607, 613, 617, 619, 631, 641, 643, 647, 653, 659, 661, 673, 677, 683, 691, 701, 709, 719, 727, 733, 739, 743, 751, 757, 761, 769, 773, 787, 797, 809, 811, 821, 823, 827, 829, 839, 853, 857, 859, 863, 877, 881, 883, 887, 907, 911, 919, 929, 937, 941, 947, 953, 967, 971, 977, 983, 991, 997]
```
Comment (pvcarrera) [inline exercises/sieve/sieve_test.rb]: This generated array is displayed in one line. Should we generate a formatted array with line breaks?

Comment (kotp) [inline exercises/sieve/sieve_test.rb]: Yes, please.

**Comment by kotp** [comment]
Hopefully can squash these commits to something logical.

**Comment by pvcarrera** [comment]
@kotp sure, I've merged master, removed the old `generate-sieve` and squashed the previous commits into one.

File: lib/sieve_cases.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class SieveCase < OpenStruct
+  OPEN_ARRAY = "[\n\s\s\s\s\s\s".freeze
+  CLOSE_ARRAY = "\n\s\s\s\s]".freeze
+  NEW_ARRAY_ROW = ",\n\s\s\s\s\s\s".freeze
+  ARRAY_ELEMENTS_PER_ROW = 17.freeze
```
Comment (Insti) [inline lib/sieve_cases.rb]: Why is 17 the right number?

Comment (pvcarrera) [inline lib/sieve_cases.rb]: In the previous `sieve_test.rb`, the `test_primes_up_to_1000` expected array had 17 elements in its first row, so I took this as the base number. The ideal solution should probably count characters and match with Rubocop rules but I wasn't sure how to do it, so I came up with this solution.

**Comment by Insti** [comment]
Thanks for your work on this @pvcarrera :heart:

**Pull request merged**